### ISSUE TITLE
pytest-cov with pytest-xdist -f == "Error in sys.exitfunc:"

### ISSUE THREAD

**Issue by diefans** [schlamar/pytest-cov#47]
I get an error when running pytest-cov together with pytest-xdist: `py.test --cov pkg -f`

The error is just
```
Error in sys.exitfunc:
```

```
$ pip freeze | grep pytest
pytest==2.6.4
pytest-cov==1.8.1
pytest-xdist==1.11
```

Everything works as expected in `pytest-cov==1.6`

In `pytest-cov==1.7.0` the exception is more explicit:
```python-traceback
Traceback (most recent call last):
  File "/home/olli/.virtualenvs/ftr/bin/py.test", line 9, in <module>
    load_entry_point('pytest==2.6.4', 'console_scripts', 'py.test')()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/config.py", line 41, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 413, in __call__
    return self._docall(methods, kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 424, in _docall
    res = mc.execute()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 315, in execute
    res = method(**kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/xdist/plugin.py", line 58, in pytest_cmdline_main
    looponfail_main(config)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/xdist/looponfail.py", line 20, in looponfail_main
    remotecontrol.loop_once()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/xdist/looponfail.py", line 87, in loop_once
    result = self.runsession()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/xdist/looponfail.py", line 76, in runsession
    return self.channel.receive()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/execnet/gateway_base.py", line 711, in receive
    raise self._getremoteerror() or EOFError()
execnet.gateway_base.RemoteError: Traceback (most recent call last):
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/execnet/gateway_base.py", line 1043, in executetask
    function(channel, **kwargs)
  File "<remote exec>", line 20, in init_slave_session
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/xdist/looponfail.py", line 174, in main
    self.config.hook.pytest_cmdline_main(config=self.config)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 413, in __call__
    return self._docall(methods, kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 424, in _docall
    res = mc.execute()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 315, in execute
    res = method(**kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/main.py", line 116, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/main.py", line 109, in wrap_session
    exitstatus=session.exitstatus)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 413, in __call__
    return self._docall(methods, kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 424, in _docall
    res = mc.execute()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 315, in execute
    res = method(**kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/terminal.py", line 349, in pytest_sessionfinish
    __multicall__.execute()
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/_pytest/core.py", line 315, in execute
    res = method(**kwargs)
  File "/home/olli/.virtualenvs/ftr/local/lib/python2.7/site-packages/pytest_cov.py", line 125, in pytest_sessionfinish
    self.cov_controller.finish()
AttributeError: 'NoneType' object has no attribute 'finish'

Error in sys.exitfunc:
```

**Comment by schlamar** [comment]
#45 should be the fix for this

**Comment by schlamar** [comment]
Fixed in #49
